ci/gh-experiment: sharness fail during sharness-html generation

Seems that sometimes successful run of sharness tests fail on Github Action due to Java error around  `test/sharness/test-results/sharness.xml`:

https://github.com/ipfs/kubo/actions/runs/3920501209/jobs/6702175413#step:10:70208:
```
*** test/sharness/aggregate ***
fixed   0
success 7711
failed  0
broken  18
total   7751
*** test/sharness/test-results/sharness.xml ***
Exception in thread "main" java.lang.NoClassDefFoundError: org/xmlresolver/Resolver
	at net.sf.saxon.lib.CatalogResourceResolver.<init>(CatalogResourceResolver.java:46)
*** test/sharness/test-results/sharness-html ***
Downloading SaxonHE11-4J.zip
Removing old lib/dependencies/SaxonHE11-4J
Unzipping lib/dependencies/SaxonHE11-4J.zip
	at net.sf.saxon.Configuration.init(Configuration.java:367)
	at net.sf.saxon.Configuration.<init>(Configuration.java:230)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.lang.Class.newInstance(Class.java:584)
	at net.sf.saxon.Configuration.newConfiguration(Configuration.java:246)
	at net.sf.saxon.s9api.Processor.<init>(Processor.java:69)
	at net.sf.saxon.Transform.doTransform(Transform.java:321)
	at net.sf.saxon.Transform.main(Transform.java:81)
Caused by: java.lang.ClassNotFoundException: org.xmlresolver.Resolver
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 12 more
make: *** [test/sharness/Rules.mk:52: test/sharness/test-results/sharness-html] Error 1
```

@galargh I had to time to look into this –  assuming you may be better person to diagnose?

--- a/test/sharness/Rules.mk
+++ b/test/sharness/Rules.mk
@@ -47,12 +47,17 @@ $(d)/test-results/sharness.xml: $(T_$(d))
 	@(cd $(@D)/.. && ./lib/test-aggregate-junit-reports.sh)
 .PHONY: $(d)/test-results/sharness.xml
 
-$(d)/test-results/sharness-html: $(d)/test-results/sharness.xml
+$(d)/download-saxon:
+	@echo "*** $@ ***"
+	@(cd $(@D) && ./lib/download-saxon.sh)
+.PHONY: $(d)/download-saxon
+
+$(d)/test-results/sharness-html: $(d)/test-results/sharness.xml $(d)/download-saxon
 	@echo "*** $@ ***"
 	@(cd $(@D)/.. && ./lib/test-generate-junit-html.sh frames)
 .PHONY: $(d)/test-results/sharness-html
 
-$(d)/test-results/sharness.html: $(d)/test-results/sharness.xml
+$(d)/test-results/sharness.html: $(d)/test-results/sharness.xml $(d)/download-saxon
 	@echo "*** $@ ***"
 	@(cd $(@D)/.. && ./lib/test-generate-junit-html.sh no-frames)
 .PHONY: $(d)/test-results/sharness.html

--- a/test/sharness/lib/download-saxon.sh
+++ b/test/sharness/lib/download-saxon.sh
@@ -1,0 +1,39 @@
+#!/bin/bash
+
+dependencies=(
+  "url=https://sourceforge.net/projects/saxon/files/Saxon-HE/11/Java/SaxonHE11-4J.zip;md5=8a4783d307c32c898f8995b8f337fd6b"
+  "url=https://raw.githubusercontent.com/pl-strflt/ant/c781f7d79b92cc55530245d9554682a47f46851e/src/etc/junit-frames-saxon.xsl;md5=6eb013566903a91e4959413f6ff144d0"
+  "url=https://raw.githubusercontent.com/pl-strflt/ant/c781f7d79b92cc55530245d9554682a47f46851e/src/etc/junit-noframes-saxon.xsl;md5=8d54882d5f9d32a7743ec675cc2e30ac"
+)
+
+dependenciesdir="lib/dependencies"
+mkdir -p "$dependenciesdir"
+
+get_md5() {
+  md5sum "$1" | cut -d ' ' -f 1
+}
+
+for dependency in "${dependencies[@]}"; do
+  url="$(echo "$dependency" | cut -d ';' -f 1 | cut -d '=' -f 2)"
+  md5="$(echo "$dependency" | cut -d ';' -f 2 | cut -d '=' -f 2)"
+  filename="$(basename "$url")"
+  if test -f "$dependenciesdir/$filename" && test "$(get_md5 "$dependenciesdir/$filename")" = "$md5"; then
+    echo "Using cached $filename"
+  else
+    echo "Downloading $filename"
+    curl -L --max-redirs 5 --retry 5 --no-progress-meter --output "$dependenciesdir/$filename" "$url"
+    actual_md5="$(get_md5 "$dependenciesdir/$filename")"
+    if test "$actual_md5" != "$md5"; then
+      echo "Downloaded $filename has wrong md5sum ('$actual_md5' != '$md5')"
+      exit 1
+    fi
+    dirname=${filename%.*}
+    extension=${filename#$dirname.}
+    if test "$extension" = "zip"; then
+      echo "Removing old $dependenciesdir/$dirname"
+      rm -rf "$dependenciesdir/$dirname"
+      echo "Unzipping $dependenciesdir/$filename"
+      unzip "$dependenciesdir/$filename" -d "$dependenciesdir/$dirname"
+    fi
+  fi
+done

--- a/test/sharness/lib/test-generate-junit-html.sh
+++ b/test/sharness/lib/test-generate-junit-html.sh
@@ -1,43 +1,5 @@
 #!/bin/bash
 
-dependencies=(
-  "url=https://sourceforge.net/projects/saxon/files/Saxon-HE/11/Java/SaxonHE11-4J.zip;md5=8a4783d307c32c898f8995b8f337fd6b"
-  "url=https://raw.githubusercontent.com/pl-strflt/ant/c781f7d79b92cc55530245d9554682a47f46851e/src/etc/junit-frames-saxon.xsl;md5=6eb013566903a91e4959413f6ff144d0"
-  "url=https://raw.githubusercontent.com/pl-strflt/ant/c781f7d79b92cc55530245d9554682a47f46851e/src/etc/junit-noframes-saxon.xsl;md5=8d54882d5f9d32a7743ec675cc2e30ac"
-)
-
-dependenciesdir="lib/dependencies"
-mkdir -p "$dependenciesdir"
-
-get_md5() {
-  md5sum "$1" | cut -d ' ' -f 1
-}
-
-for dependency in "${dependencies[@]}"; do
-  url="$(echo "$dependency" | cut -d ';' -f 1 | cut -d '=' -f 2)"
-  md5="$(echo "$dependency" | cut -d ';' -f 2 | cut -d '=' -f 2)"
-  filename="$(basename "$url")"
-  if test -f "$dependenciesdir/$filename" && test "$(get_md5 "$dependenciesdir/$filename")" = "$md5"; then
-    echo "Using cached $filename"
-  else
-    echo "Downloading $filename"
-    curl -L --max-redirs 5 --retry 5 --no-progress-meter --output "$dependenciesdir/$filename" "$url"
-    actual_md5="$(get_md5 "$dependenciesdir/$filename")"
-    if test "$actual_md5" != "$md5"; then
-      echo "Downloaded $filename has wrong md5sum ('$actual_md5' != '$md5')"
-      exit 1
-    fi
-    dirname=${filename%.*}
-    extension=${filename#$dirname.}
-    if test "$extension" = "zip"; then
-      echo "Removing old $dependenciesdir/$dirname"
-      rm -rf "$dependenciesdir/$dirname"
-      echo "Unzipping $dependenciesdir/$filename"
-      unzip "$dependenciesdir/$filename" -d "$dependenciesdir/$dirname"
-    fi
-  fi
-done
-
 case "$1" in
   "frames")
     java -jar lib/dependencies/SaxonHE11-4J/saxon-he-11.4.jar \

